### ISSUE TITLE
Clashing with some Fabric Mods under 1.18

### ISSUE THREAD

**Issue by UntalentedAmateur** [GameParrot/minecraft-mac-window-fix#3]
This has been an excellent tool the whole time I played under 1.17, but with the update to 1.18 I had a variety of odd issues crop up.

To begin with, I couldn't get it to install cleanly, with Minecraft crashing before even getting to the Red Mojang screen, but eventually trashed the prior install and tried again - success! (with the chmod+x as necesary to get the install script to run).

Second issue is that the dock icons for the Java 16 and Java 17 Minecraft versions seemed to be missing - just a blank application icon.

Third issue was that I started getting unexplained crashes (seemingly related to programming of fabric mods) when loading into a Paper 1.18 multiplayer server, and the same crash when loading a known good 1.18 singleplayer world.

I was able to isolate the crashes to the lambdynamiclights-2.1.0+1.17 and the plasmovoice-fabric-1.18-1.2.7 mods, but as soon as I removed the manual Java executable and used the default Minecraft supplied one, all mods worked fine (except the microphone support for plasmovoice, as expected) and all worlds were accessible.

I suspect there's an issue with what / how the versions of Java were installed, as that seems to be what's tripping the mods up.  And, combined with the initial issues getting it to cleanly install, I suspect something might have gotten corrupted.

**Comment by GameParrot** [comment]
Try reinstalling it, which will fix corruptions. To reinstall it, just run the installation command again.

**Comment by GameParrot** [comment]
Also, make sure you use the correct Java version. It gives you a path for Java 17, a path for Java 16, and a path for Java 8. For 1.18 and above, use Java 17 (the first path it gives you). For 1.17, use Java 16 (the 2nd path). For versions older than 1.17, use Java 8 (the last path).

**Comment by UntalentedAmateur** [comment]
Hi,

I'm sure I'm using the correct Java version path, but anyway I found a solution through the modification of MinecraftLaunch.  If you look at the attached, you'll see what I've stripped out, and it worked quite happily -- keeping the rest of the improvements in place (i.e. microphone support, etc), and the Fabric mods stopped complaining about inconsistencies with the default installation.  I've been able to get it to work with the modified file across four differently configured computers without issue, so I'm happy with my modification.

When I installed, I downloaded the latest installer.sh from your releases and ran it from there.  Since I could see what it was doing, I was able to manually uninstall the old installation when I first had issues.

[MinecraftLaunch.zip](https://github.com/GameParrot/minecraft-mac-window-fix/files/7655331/MinecraftLaunch.zip)
